src/devices/Ad7193/Ad7193.cs

```diff
@@ -0,0 +1,842 @@
+        /// </summary>
+        public void Calibrate()
+        {
+            // internal zero scale calibration (MD2 = 1, MD1 = 0, MD0 = 0)
+            SetRegisterBits(Register.Mode, BitMask.ModeMD, 0b100);
+            WaitForADC();
+
+            // internal full scale calibration (MD2 = 1, MD1 = 0, MD0 = 1)
+            SetRegisterBits(Register.Mode, BitMask.ModeMD, 0b101);
+            WaitForADC();
+        }
+
+        /// <summary>
+        /// Waits after a single conversion until the DOUT/!RDY goes low to indicate the completion of a conversion
+        /// </summary>
+        public void WaitForADC()
+        {
+            while (!IsReady)
+            {
+                ReadRegisterValue(Register.Status);
+                Thread.Sleep(5);
+            }
+        }
+
+        /// <summary>
+        /// Initiates Single Conversion (device will go into low power mode when conversion complete, and DOUT/!RDY goes low to indicate the completion of a conversion)
+        /// </summary>
+        public void StartSingleConversion()
+        {
+            // set the single conversion mode bits
+            SetRegisterBits(Register.Mode, BitMask.ModeMD, 0b001);
+
+            _stopWatch.Restart();
+        }
+
+        /// <summary>
+        /// Initiates Continuous Conversion. After calling this method the AdcValueReceived event will be fired continuously. This mode is much faster, but uses significatily more CPU power.
+        /// </summary>
+        /// <param name="channels">The channel(s) on which the conversion is taking place.</param>
+        /// <param name="frequency">The target frequency of the sampling. AD7193 has the maximum of 4800 samples per second.</param>
+        public void StartContinuousConversion(Channel channels, uint frequency = 0)
+        {
+            ActiveChannels = channels;
+
+            if (frequency == 0)
+            {
+                frequency = metadata.ADCSamplerate;
+            }
+
+            if (frequency > metadata.ADCSamplerate)
+            {
+                throw new ArgumentException($"The frequency you provided is higher than the maximum sampling rate of AD7193 ({metadata.ADCSamplerate} SPS).");
+            }
+
+            // set the continuous conversion mode bits
+            SetRegisterBits(Register.Mode, BitMask.ModeMD, 0b000);
+
+            ContinuousRead = true;
+
+            long samplePerTicks = Stopwatch.Frequency / frequency;
+            if (samplePerTicks == 0)
+            {
+                samplePerTicks = 1;
+            }
+
+            _stopWatch.Restart();
+            Task samplingTask = Task.Run(() =>
+            {
+                long samples = 0;
+                long nextSampleAt = _stopWatch.ElapsedTicks;
+                long elapsedTicks = 0;
+                long jitter = 0;
+                long maxJitterCorrectionPerSample = Math.Max(Stopwatch.Frequency / 100000, 1);
+                while (_stopWatch.IsRunning)
```

Comment:
How does user get value here? How does it know it got updated? How does user stop this loop? Should you create an event which you call in here? I'd prefer we didn't have any threads in the binding - it's usually better to add helper methods after the binding is complete